From stable+bounces-300015-greg=kroah.com@vger.kernel.org Wed Aug 12 02:10:16 2026
From: Sasha Levin <sashal@kernel.org>
Date: Tue, 11 Aug 2026 20:09:31 -0400
Subject: include/linux/fs.h: add inode_lock_killable()
To: stable@vger.kernel.org
Cc: Max Kellermann <max.kellermann@ionos.com>, Christian Brauner <brauner@kernel.org>, Sasha Levin <sashal@kernel.org>
Message-ID: <20260812000933.4158055-1-sashal@kernel.org>

From: Max Kellermann <max.kellermann@ionos.com>

[ Upstream commit d8c5507cd140d9471472ece673e70250b957c595 ]

Prepare for making inode operations killable while they're waiting for
the lock.

Signed-off-by: Max Kellermann <max.kellermann@ionos.com>
Link: https://lore.kernel.org/20250513150327.1373061-1-max.kellermann@ionos.com
Signed-off-by: Christian Brauner <brauner@kernel.org>
Stable-dep-of: fa724e235cfd ("cifs: add fscache_resize_cookie() to cifs_setsize()")
Signed-off-by: Sasha Levin <sashal@kernel.org>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 include/linux/fs.h |   10 ++++++++++
 1 file changed, 10 insertions(+)

--- a/include/linux/fs.h
+++ b/include/linux/fs.h
@@ -848,6 +848,11 @@ static inline void inode_lock(struct ino
 	down_write(&inode->i_rwsem);
 }
 
+static inline __must_check int inode_lock_killable(struct inode *inode)
+{
+	return down_write_killable(&inode->i_rwsem);
+}
+
 static inline void inode_unlock(struct inode *inode)
 {
 	up_write(&inode->i_rwsem);
@@ -858,6 +863,11 @@ static inline void inode_lock_shared(str
 	down_read(&inode->i_rwsem);
 }
 
+static inline __must_check int inode_lock_shared_killable(struct inode *inode)
+{
+	return down_read_killable(&inode->i_rwsem);
+}
+
 static inline void inode_unlock_shared(struct inode *inode)
 {
 	up_read(&inode->i_rwsem);
